src/App.js

```diff
@@ -0,0 +1,149 @@
+import React, { useState } from 'react';
+
+function App() {
+  return (
+    <main>
+      <NoughtsCrosses />
+    </main>
+  );
+}
+
+function NoughtsCrosses() {
+  const defaultGame = {
+    board: Array(9).fill(null),
+    isXNext: true,
+    history: [Array(9).fill(null)],
+  };
+  const [game, setGame] = useState(defaultGame);
+  const { board, isXNext, history } = game;
+  const winner = getWinner(board);
+
+  const handlePlay = (i) => {
+    if (winner || board[i]) return;
+    const newBoard = [...board];
+    newBoard[i] = isXNext ? 'X' : 'O';
+    setGame({
+      ...game,
+      board: newBoard,
+      isXNext: !isXNext,
+      history: [...history, [...newBoard]],
+    });
+  };
+
+  const newGame = () => setGame(defaultGame);
+
+  return (
+    <>
+      <h1 id="title">Noughts and Crosses</h1>
+      <ul className="actions">
+        <li>
+          <button className="action" onClick={newGame}>
+            New Game
+          </button>
+        </li>
+      </ul>
+      <p className="status">
+        Next Player:
+        {isXNext ? <strong> X</strong> : <strong> O</strong>}
+      </p>
+      {winner && (
+        <p className="status">
+          The winner is
+          {winner === 'X' ? <strong> X</strong> : <strong> O</strong>}
+          !
+        </p>
+      )}
+      <Board board={board} handlePlay={handlePlay} />
+      <History history={history} game={game} setGame={setGame} />
+    </>
+  );
+}
+
+function Board({ board, handlePlay }) {
```

Comment:
Could you talk more about reusability? Does it refer to reusing components for different projects? What makes components reusable? Thanks!

Reply:
Good question! Re-use probably wasn't the right word, but in this case the `Board` component is pretty flexible. You pass it an array of values and it will render out the game board for you. This means that, when you use the history component to change what the board should look like, it just re-renders itself. So I guess this is more about separation of concerns than re-usability. This component knows about just the right amount of stuff. Let me know if that clarifies things a little bit!